### PR TITLE
CPP style NDR formatting

### PR DESCRIPTION
**Pull request by irsl**
The PR I'm submitting about OVDN relies on this change.

### CODE REVIEW

**Comment by irsl** [comment]
Has the CLA sign off any propagation delays? I've submitted that like 15 minutes ago.

**Comment by irsl** [comment]
@googlebot I fixed it.

File: NtApiDotNet/Ndr/NdrFormatter.cs [hunk 1]
```diff
@@ -228,7 +382,77 @@ public static INdrFormatter Create(IDictionary<Guid, string> iids_to_names, Func
...
+
+        /// <summary>
+        /// Create the default formatter.
+        /// </summary>
+        /// <param name="iids_to_names">Specify a dictionary of IIDs to names.</param>
+        /// <param name="flags">Formatter flags.</param>
+        /// <returns>The default formatter.</returns>
+        public static INdrFormatter Create(IDictionary<Guid, string> iids_to_names, DefaultNdrFormatterFlags flags)
+        {
+            return Create(iids_to_names, flags);
```
Comment (tyranid) [inline NtApiDotNet/Ndr/NdrFormatter.cs]: Maybe I'm missing something but doesn't this function just call itself?

File: NtApiDotNet/Ndr/NdrFormatter.cs [hunk 2]
```diff
@@ -238,7 +462,7 @@ public static INdrFormatter Create(IDictionary<Guid, string> iids_to_names, Defa
         /// <returns>The default formatter.</returns>
         public static INdrFormatter Create(IDictionary<Guid, string> iids_to_names)
         {
-            return Create(iids_to_names, s => s);
+            return Create(iids_to_names);
```
Comment (tyranid) [inline NtApiDotNet/Ndr/NdrFormatter.cs]: Maybe I'm missing something but doesn't this function just call itself?

File: NtApiDotNet/Ndr/NdrFormatter.cs [hunk 3]
```diff
@@ -76,12 +93,13 @@ internal string IidToName(Guid iid)
             return null;
         }
 
-        internal string DemangleComName(string name)
+        string INdrFormatterInternal.DemangleComName(string name)
         {
+            if (_demangle_com_name == null) return name;
```
Comment (tyranid) [inline NtApiDotNet/Ndr/NdrFormatter.cs]: Why add this check here? Perhaps should add it to the constructor and replace a null _demangle_com_name with the identity function s => s.

File: NtApiDotNet/Ndr/NdrFormatter.cs [hunk 4]
```diff
@@ -228,7 +382,77 @@ public static INdrFormatter Create(IDictionary<Guid, string> iids_to_names, Func
...
+        }
+
+        /// <summary>
+        /// Create the default formatter.
+        /// </summary>
+        /// <param name="iids_to_names">Specify a dictionary of IIDs to names.</param>
+        /// <returns>The default formatter.</returns>
+        public static INdrFormatter Create(IDictionary<Guid, string> iids_to_names)
+        {
+            return Create(iids_to_names);
```
Comment (tyranid) [inline NtApiDotNet/Ndr/NdrFormatter.cs]: Maybe I'm missing something but doesn't this function just call itself?

File: NtApiDotNet/Ndr/NdrFormatter.cs [hunk 5]
```diff
@@ -228,7 +382,77 @@ public static INdrFormatter Create(IDictionary<Guid, string> iids_to_names, Func
         /// <returns>The default formatter.</returns>
         public static INdrFormatter Create(IDictionary<Guid, string> iids_to_names, DefaultNdrFormatterFlags flags)
         {
-            return Create(iids_to_names, s => s, flags);
+            return Create(iids_to_names, flags);
```
Comment (tyranid) [inline NtApiDotNet/Ndr/NdrFormatter.cs]: Maybe I'm missing something but doesn't this function just call itself?

File: NtApiDotNet/Ndr/NdrFormatter.cs (hunk 3 above)
Comment (irsl) [reply NtApiDotNet/Ndr/NdrFormatter.cs]: I didn't like having the same s =>s lambda twice and also tried avoiding potential null pointer exceptions via this refactoring. It indeed broke the other constructor calls you spotted...
I'm resolving your requests the coming days.

File: NtApiDotNetTest/Ndr/NdrTest.cs [hunk 6]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
```
[the remark this answers is not in the document]
Comment (irsl) [reply NtApiDotNetTest/Ndr/NdrTest.cs]: I thought my change won't be accepted without a test (Beyoncé...), so I added the stub and was expecting some hints. 
I'm removing these files as you requested.

[the remark this answers is not in the document]
Comment (tyranid) [reply NtApiDotNetTest/Ndr/NdrTest.cs]: Yeah I've yet to implement unit testing, maybe for v2 :-)

**Comment by tyranid** [comment]
LGTM. Although it's still complaining about the CLA. You need to ensure you've signed the CLA and mark the patches with the email address you specified in the CLA.

**Comment by irsl** [comment]
@googlebot I fixed it.

**Comment by irsl** [comment]
@tyranid : Yeah, and the complaint was actually fully correct; I had an old email address configured. It is now fixed for real.
Thanks,
Imre

**Comment by tyranid** [comment]
Committed. Give me a day or so to make a release of NtApiDotNet to update the OVDN package.

**Pull request merged**